tests/integration/release/test_release.py

```diff
@@ -166,3 +175,139 @@ def test_unnoised_id_cols(dataset_name: str, request: FixtureRequest) -> None:
         .all()
         .all()
     )
+
+
+@pytest.mark.parametrize(
+    "probabilities",
+    [
+        {
+            "row_probability_in_households_under_18": 1.0,
+            "row_probability_in_college_group_quarters_under_24": 1.0,
+        },
+        {
+            "row_probability_in_households_under_18": 0.7,
+            "row_probability_in_college_group_quarters_under_24": 0.8,
+        },
+    ],
+)
+def test_guardian_duplication(
+    dataset_params: tuple[str | int | Callable[..., pd.DataFrame] | None, ...],
+    dataset_name: str,
+    probabilities: dict[str, float],
+    fuzzy_checker: FuzzyChecker,
+    mocker: MockerFixture,
+) -> None:
+    if dataset_name != DatasetNames.CENSUS:
+        return
+
+    # patch these to avoid updating dtypes and dropping columns we need for testing
+    mocker.patch("pseudopeople.dataset.coerce_dtypes", side_effect=lambda df, _: df)
+    mocker.patch(
+        "pseudopeople.dataset.Dataset.drop_non_schema_columns", side_effect=lambda df, _: df
+    )
+    # allow all irrelevant probabilities to be 0 in our config
+    mocker.patch(
+        "pseudopeople.configuration.generator.validate_overrides",
+        side_effect=lambda *args: None,
+    )
+    # allow our noise levels to be high in testing
+    mocker.patch(
+        "pseudopeople.configuration.generator.validate_noise_level_proportions",
+        lambda *args: None,
+    )
+
+    # get unnoised data
+    _, _, source, year, state, engine = dataset_params
+    unnoised = generate_decennial_census(
+        source=source, config=NO_NOISE, year=year, state=state, engine=engine
+    )
+
+    # get noised data using custom config
+    config_dict = get_single_noise_type_config(
+        dataset_name, NOISE_TYPES.duplicate_with_guardian.name
+    )
+    for probability_key in [
+        Keys.ROW_PROBABILITY_IN_COLLEGE_GROUP_QUARTERS_UNDER_24,
+        Keys.ROW_PROBABILITY_IN_HOUSEHOLDS_UNDER_18,
+    ]:
+        config_dict[dataset_name][Keys.ROW_NOISE][NOISE_TYPES.duplicate_with_guardian.name][
+            probability_key
+        ] = probabilities[probability_key]
+    config = NoiseConfiguration(LayeredConfigTree(config_dict))
+    noised = generate_decennial_census(
+        source=source, config=config.to_dict(), year=year, state=state, engine=engine
+    )
+
+    duplicated = noised.loc[noised["simulant_id"].duplicated()]
+    duplicated["age"] = duplicated["age"].astype(int)
+
+    # separate tests for household under 18 and for college under 24
+    for probability_name, age, housing_type in zip(
+        [
+            Keys.ROW_PROBABILITY_IN_HOUSEHOLDS_UNDER_18,
+            Keys.ROW_PROBABILITY_IN_COLLEGE_GROUP_QUARTERS_UNDER_24,
+        ],
+        [18, 24],
+        ["Household", "College"],
+    ):
+        group_data = unnoised.loc[
+            (unnoised["age"].astype(int) < age)
+            & (unnoised["housing_type"] == housing_type)
+            & (unnoised["guardian_1"].notna())
```

Comment:
because guardian_1 is never nan, right? But guardian_2 might be?

Reply:
Yes if "guardian_1" is notna then there is at least one guardian but not necessarily guardian 2